lib/kino/process/tracer.ex

```diff
@@ -31,10 +31,28 @@ defmodule Kino.Process.Tracer do
       message: message
     }
 
-    {:noreply, [new_event | trace_events]}
+    trace_events = [new_event | trace_info.raw_trace_events]
+
+    process_labels =
+      trace_info.process_labels
+      |> put_new_label(from)
+      |> put_new_label(to)
+
+    {:noreply, %{trace_info | raw_trace_events: trace_events, process_labels: process_labels}}
   end
 
   def handle_info(_ignored_event, trace_events) do
     {:noreply, trace_events}
   end
+
+  defp put_new_label(process_labels, pid) do
+    Map.put_new(process_labels, pid, get_label(pid))
```

Comment:
```suggestion
    Map.put_new_lazy(process_labels, pid, fn -> get_label(pid) end)
```
This way we only compute the label if necessary.